generic.go

```diff
@@ -148,3 +148,33 @@ func Run(ctx context.Context, img string, opts ...ContainerCustomizer) (*DockerC
 
 	return c, nil
 }
+
+// GetContainerHost gets host (IP or name) of the docker daemon where container port is exposed.
+// Note: prefer using Container.Host when it is possible.
+// Warning: this function is based on Docker host setting, it will fail if using an SSH tunnel.
+// "TESTCONTAINERS_HOST_OVERRIDE" env variable can be used to customize host returned by this function.
+func GetContainerHost(ctx context.Context, opts ...ContainerCustomizer) (string, error) {
```

Comment:
suggestion: it's idiomatic to avoid Get prefix used by other languages unless there's a conflict.

@mdelapenya how does this play with the incoming context support, are we introducing something that will be redundant or need changing for that?

I suspect this was introduced in response to PR feedback, but its bloating the surface area of this, might be better to do it as a follow up, @mdelapenya thoughts?

Reply:
Renamed to `ContainerHost`.